src/App.tsx

```diff
@@ -1,14 +1,54 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos, getVisibleTodos } from './api';
+import { Todo } from './types/Todo';
+import { TodoStatus } from './components/TodoStatus/TodoStatus';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedValue, setSelectedValue] = useState(TodoStatus.All);
+  const [query, setQuery] = useState('');
+  const [todoModal, setTodoModal] = useState<Todo | null>(null);
+  const [isModalButtonClicked, setIsModalButtonClicked] = useState(false);
+
+  useEffect(() => {
+    const fetchTodosFromAPI = async () => {
+      try {
+        setTodos(await getTodos());
+      } catch (error) {
+        throw new Error('Error fetching todos');
```

Comment:
Instead of throwing a new error here, consider showing an error message or a notification to the user, so they can know what went wrong.